library/std/src/sys/windows/process.rs

```diff
@@ -326,7 +326,13 @@ fn resolve_exe<'a>(exe_path: &'a OsStr) -> io::Result<Vec<u16>> {
             &"program path has no file name",
         ));
     }
-    let has_exe_suffix = exe_path.bytes().ends_with(b".exe");
+    // Test if the file name has the `exe` extension.
+    // This does a case-insensitive `ends_with`.
+    let has_exe_suffix = exe_path
+        .bytes()
+        .get(exe_path.len() - EXE_SUFFIX.len()..)
```

Comment:
This subtraction overflows on short `exe_path`.

Reply:
Thanks, that was a silly mistake. I've added new tests in-tree.